src/otx/algo/utils/xai_utils.py

```diff
@@ -195,3 +230,51 @@ def _get_overlay(img: np.ndarray, s_map: np.ndarray, weight: float = 0.3) -> np.
     overlay = img * weight + s_map * (1 - weight)
     overlay[overlay > 255] = 255
     return overlay.astype(np.uint8)
+
+
+def _crop_padded_map(
+    batch_saliency_map: list[dict[Any, np.array]],
+    image_shape: tuple[int, int],
+    paddings: list[tuple[int, int, int, int]],
+) -> list[dict[Any, np.array]]:
+    """Crop padded map."""
+    # Padding: number of pixels to pad all borders (left, top, right, bottom)
+    height, width = image_shape
+
+    for i, image_map in enumerate(batch_saliency_map):
+        left, top, right, bottom = paddings[i]
+        left_ratio, top_ratio, right_ratio, bottom_ratio = left / width, top / height, right / width, bottom / height
+        for class_idx, class_map in image_map.items():
+            map_h, map_w = class_map.shape
+            d_top, d_bottom = int(top_ratio * map_h), int(bottom_ratio * map_h)
+            d_left, d_right = int(left_ratio * map_w), int(right_ratio * map_w)
```

Comment:
Why don't you use `round()`? Or, we can even adopt bi-linear interpolation, like ROIAlign in MaskRCNN, to be more precise. (in the future)

Reply:
I followed the same approach we have in merging saliency maps for tiling where I also had to crop maps:

https://github.com/openvinotoolkit/training_extensions/blob/785a560dbff160d2a25e64ca114bd48fb943a632/src/otx/core/utils/tile_merge.py#L258-L259

Reply:
The coordinates will be shifted to left a bit due to this "floor" operation.
But let's keep it as is for consistency or backward compatibility.